src/galois_16.rs

```diff
@@ -0,0 +1,327 @@
+                }
+            }
+        }
+
+        let remainder = Element::constant(poly[2]);
+        let quotient = Element([poly[0], poly[1]]) * leading_mul_inv;
+
+        (quotient, remainder)
+    }
+
+    fn polynom_div(self, rhs: Self) -> (Element, Element) {
+        let divisor_degree = rhs.degree();
+        if rhs.is_zero() {
+            panic!("divide by 0");
+        } else if self.degree() < divisor_degree {
+            // If divisor's degree (len-1) is bigger, all dividend is a remainder
+            (Element::zero(), self)
+        } else if divisor_degree == 0 {
+            // divide by constant.
+            let invert = galois_8::div(1, rhs.0[1]);
+            let quotient = Element([
+                galois_8::mul(invert, self.0[0]),
+                galois_8::mul(invert, self.0[1]),
+            ]);
+
+            (quotient, Element::zero())
+        } else {
+            // self degree is at least divisor degree, divisor degree not 0.
+            // therefore both are 1. 
+            debug_assert_eq!(self.degree(), divisor_degree);
+            debug_assert_eq!(self.degree(), 1);
+
+            // ensure rhs is constant.
+            let leading_mul_inv = galois_8::div(1, rhs.0[0]);
+            let monic = Element([
+                galois_8::mul(leading_mul_inv, rhs.0[0]),
+                galois_8::mul(leading_mul_inv, rhs.0[1]),
+            ]);
+
+            let leading_coeff = self.0[0];
+            let mut remainder = self.0[1];
+
+            if monic.0[1] != 0 {
+                remainder ^= galois_8::mul(monic.0[1], self.0[0]);
+            }
+
+            (
+                Element::constant(galois_8::mul(leading_mul_inv, leading_coeff)),
+                Element::constant(remainder),
+            )
+        }
+    }
+
+    /// Convert the inverse of this field element. Panics if zero.
+    fn inverse(self) -> Element {
+        if !self.is_zero() {
+            // first step of extended euclidean algorithm.
+            // done here because EXT_POLY is outside the scope of `Element`.
+            let (gcd, y) = {
+                // self / EXT_POLY = (0, self)
+                let remainder = self; 
+
+                // GCD is constant because EXT_POLY is irreducible
+                let (g, x, _) = remainder.const_egcd(EgcdRhs::ExtPoly);
+
+                (g, x)
+            };
+
+            // we still need to normalize it by dividing by the gcd
+            if gcd != 0 {
+                // EXT_POLY is irreducible so the GCD will always be constant.
+                // EXT_POLY*x + self*y = gcd
+                // self*y = gcd - EXT_POLY*x
+                //
+                // EXT_POLY*x is representative of the equivalence class of 0.
+                let normalizer = galois_8::div(1, gcd);
+                return y * normalizer;
+            }
+        }
+        
+        // either self is zero polynomial or is equivalent to 0
+        panic!("Cannot invert 0");
```

Comment:
Personally I feel it would be nicer to move the `panic!(...)` to top the top of the code, or change this function into `if ... else ..` branching, as these two options feel easier for audit and analysis.

Just to be more specific, moving it to top would be
```
if self.zero(...) {
    panic!("Cannot invert 0");
}

// first step of extended euclidean algorithm.
...

changing into `if ... else ...` branch would be
```
if self.zero(...) {
    panic!("Cannot invert 0");
} else {
    // first step of extended euclidean algorithm.
    ...
}
```

Largely my habit though, the main rationale would be regardless of function size, the `panic!()`s would always be visible at top of function.